client/lib/analytics/index.js

```diff
@@ -95,6 +95,14 @@ const analytics = {
 		_superProps = superProps;
 	},
 
+	checkDoNotTrack: function() {
+		if ( navigator.doNotTrack ) {
+			return false;
+		}
+
+		return true;
+	},
```

Comment:
Something like `shouldTrack`, `shouldTrackUser` or `shouldTrackAgent` would be clearer and more conventional. 

What I worry about is testability and dependency on browsers: `navigator` is a global that depends on the environment running. Yes, those can be mocked, but everyone testing pieces downstream that depend on this check will have to be aware of that.